src/main/webapp/assets/js/view/application-add-wizard.js

```diff
@@ -680,10 +680,12 @@ define([
                 this.$('.required-config-loading').hide()
                 if (catalogEntryItem!=null && catalogEntryItem.config!=null) {
                     var that = this
-                    _.each(catalogEntryItem.config, function (cfg) {
-                        if (cfg.priority !== undefined) {
+                    _.chain(catalogEntryItem.config).sortBy(function (cfg){
+                        return !isNaN(cfg.priority) ? cfg.priority : 99999;
+                    }).each(function (cfg) {
+                        if (cfg.pinned === true || (cfg.contraints.indexOf('required') > -1 && typeof cfg.defaultValue === 'string' && cfg.defaultValue === '')) {
```

Comment:
> Would be weird to have a defaultValue the empty string and required at the same time, but you never know :)

That's the point of this check, what if as a blueprint author I want to ensure that one of my parameter has be filled without giving it a default?